docs/source/architecture/journal.rst

```diff
@@ -0,0 +1,379 @@
+generating candidate blocks to extend the current chain. 
+
+The Journal is the consumer of Blocks and Batches that arrive at the validator.
+These Blocks and Batches arrive via interconnect, either through the gossip
+protocol or the rest-api. The newly-arrived Blocks and Batches are sent to the
+Journal, which routes them internally.
+
+.. image:: ../images/journal_organization.*
+   :width: 80%
+   :align: center
+   :alt: Journal Organization Diagram
+
+The Journal divides up the processing of Blocks and Batches to different
+processing pipelines. Both objects are delivered initially to the Completer,
+which guarantees that all dependencies for the Blocks and Batches have been
+satisfied and delivered downstream. Completed Blocks are delivered to the Chain
+controller for validation and fork resolution. Completed Batches are delivered
+the BlockPublisher for validation and inclusion in a Block.
+
+The Journal is designed to be asynchronous, allowing incoming blocks to be
+processed in parallel by the ChainController, as well as allowing the
+BlockPublisher to proceed with claiming blocks even when the incoming block
+rate is high.
+
+The Journal is also designed to be flexible enough to accept different
+consensus algorithms.  The Journal implements a consensus interface that
+defines the entry points and responsibilities of a consensus algorithm.
+
+The BlockStore
+==============
+
+The BlockStore contains all the blocks in the current block chain, that is the
+list of blocks from the current chain head back to the Genesis blocks. Blocks
+from forks are not included in the BlockStore. The BlockStore also includes a
+reference to the head of the current chain. The BlockStore is expected to be
+coherent at all times, and an error in the BlockStore is considered a
+non-recoverable error for the validator. Such critical errors would include
+missing blocks, bad indexes, missing chain reference, incomplete blocks or
+invalid blocks in the store. The BlockStore provides an atomic means to update
+the store when the current fork is changed (the chain head is updated).
+
+The BlockStore is a persistent on-disk store of all Blocks in the current
+chain. When the validator is started, the contents of the BlockStore is trusted
+to be the current state of the world. The BlockStore also stores the identifier
+of the chain head.
+
+All blocks stored here are formally complete. The BlockStore allows blocks to
+be accessed via Block ID, additionally blocks may be accessed via Batch ID or
+Transaction ID.  I.e. GetBlockByTransactionId, GetBatchByTransactionId and
+GetBlockByBatchId.
+
+The BlockStore maintains internal mappings of Transaction-to-Block and
+Batch-to-Block. These may be rebuilt if missing or corrupt, this rebuild should
+be done during start up, not during the course of normal operation. These
+mappings should be stored in a format that is cached to disk, so they are not
```

Comment:
Run-on sentence. Separate, e.g.:
"These may be rebuilt if missing or corrupt, this rebuild should be done during start up, not during the course of normal operation."
to
These may be rebuilt if missing or corrupt. This rebuild should be done during start up, not during the course of normal operation.